storefront-reference-architecture-master/cartridges/app_storefront_custom_cartridge/cartridge/controllers/DwScripts.js

```diff
@@ -0,0 +1,61 @@
+var server = require("server");
+var ProductMgr = require("dw/catalog/ProductMgr");
+var CatalogMgr = require("dw/catalog/CatalogMgr");
+var CustomerMgr = require("dw/catalog/CustomerMgr");
+/**
+ *
+ * @param {String} id
+ * @returns {dw.catalog.Product}
+ */
+const getProductById = (id) => {
+    return ProductMgr.getProduct(id);
+};
+/**
+ *
+ * @param {dw.catalog.Product} product
+ * @returns {dw.Catalog.Category}
+ */
+const getProductsCategory = (product) => {
+    return product.getPrimaryCategory();
+};
+
+/**
+ *
+ * @param {dw.catalog.Product} product
+ * @returns {dw.value.Money}
+ */
+const getProductsPrice = (product) => {
+    return product.getPriceModel().getPrice();
+};
+/**
+ *
+ * @param {String} id
+ * @returns {dw.customer.Customer}
+ */
+const getCustomerById = (id) => {
+    return CustomerMgr.getCustomerByNumber(id);
+};
+/**
+ *
+ * @param {String} catalogID
+ * @returns Category
+ */
+const getMainCategories = (catalogID) => {
+    const catalog = CatalogMgr.getCatalog(catalogID);
+    return catalog.getRoot();
```

Comment:
Will this do the job? You need to get **all** the main/primary categories.

Reply:
Will it do the job if i chain it with .getSubCategories()

Reply:
Yes, this will give you level 2 categories, which are the ones you need

Reply:
Ok thank you!